core/data/define_spec.rb

```diff
@@ -0,0 +1,54 @@
+require_relative '../../spec_helper'
+require_relative 'fixtures/classes'
+
+ruby_version_is "3.2" do
+  describe "Data.define" do
+    describe "stored as constant" do
+      it "accepts no arguments" do
+        EmptyData = Data.define
+        EmptyData.members.should == []
+        EmptyData.inspect.should == "EmptyData"
+      end
+
+      it "accepts symbols" do
+        MovieWithSymbol = Data.define(:title, :year)
+        MovieWithSymbol.members.should == [:title, :year]
+        MovieWithSymbol.inspect.should == "MovieWithSymbol"
+      end
+
+      it "accepts strings" do
+        MovieWithString = Data.define("title", "year")
+        MovieWithString.members.should == [:title, :year]
+        MovieWithString.inspect.should == "MovieWithString"
+      end
+
+      it "accepts first argument as string" do
+        BlockbusterMovie = Data.define("Blockbuster", :title, :year)
+        BlockbusterMovie.members.should == [:title, :year]
+        BlockbusterMovie.inspect.should == "Data::Blockbuster"
+      end
+    end
+
+    describe "stored as variable" do
+      it "accepts no arguments" do
+        empty_data = Data.define
+        empty_data.members.should == []
+      end
+
+      it "accepts symbols" do
+        movie_with_symbol = Data.define(:title, :year)
+        movie_with_symbol.members.should == [:title, :year]
+      end
+
+      it "accepts strings" do
+        movie_with_string = Data.define("title", "year")
+        movie_with_string.members.should == [:title, :year]
+      end
+
+      it "accepts first argument as string" do
```

Comment:
"accepts a mix of strings and symbols"
and then below check all of them are in members

Reply:
Good idea, thanks.